ssg/templates.py

```diff
@@ -120,66 +142,19 @@ def __init__(
                 output_dir = self.remediations_dir
             dir_ = os.path.join(output_dir, lang_dir)
             self.output_dirs[lang_name] = dir_
-        # scan directory structure and dynamically create list of templates
-        for item in sorted(os.listdir(self.templates_dir)):
-            maybe_template = Template(templates_dir, item)
-            if maybe_template.looks_like_template():
-                maybe_template.load()
-                templates[item] = maybe_template
-        self.product_cpes = ProductCPEs()
-        self.product_cpes.load_cpes_from_directory_tree(cpe_items_dir, self.env_yaml)
-
-    def build_lang_file(
-            self, rule_id, template_name, template_vars, lang, local_env_yaml):
-        """
-        Builds and returns templated content for a given rule for a given
-        language; does not write the output to disk.
-        """
-        if lang not in templates[template_name].langs:
-            return None
-
-        template_file_name = lang.name + ".template"
-        template_file_path = os.path.join(self.templates_dir, template_name, template_file_name)
-        template_parameters = templates[template_name].preprocess(template_vars, lang.name)
-        jinja_dict = ssg.utils.merge_dicts(local_env_yaml, template_parameters)
-        filled_template = ssg.jinja.process_file_with_macros(
-            template_file_path, jinja_dict)
 
-        return filled_template
-
-    def build_lang(
-            self, rule_id, template_name, template_vars, lang, local_env_yaml, platforms=None):
+    def get_template_backend_langs(self, template, rule_id):
         """
-        Builds templated content for a given rule for a given language.
-        Writes the output to the correct build directories.
+        Returns list of languages that should be generated from a template
+        configuration, controlled by backends.
         """
-        if lang not in templates[template_name].langs or lang.name == "sce-bash":
-            return
-
-        filled_template = self.build_lang_file(rule_id, template_name,
-                                               template_vars, lang,
-                                               local_env_yaml)
-
-        ext = lang.file_extension
-        output_file_name = rule_id + ext
-        output_filepath = os.path.join(
-            self.output_dirs[lang.name], output_file_name)
-
-        with open(output_filepath, "w") as f:
-            f.write(filled_template)
-
-    def get_langs_to_generate(self, rule):
-        """
-        For a given rule returns list of languages that should be generated
-        from templates. This is controlled by "template_backends" in rule.yml.
-        """
-        if "backends" in rule.template:
-            backends = rule.template["backends"]
+        if "backends" in template:
+            backends = template["backends"]
```

Comment:
The Code Climate now started to complain about a cognitive complexity of this method. You can probably try to extract a method from the code in the `if` block.

Reply:
Made it simpler, let's see what CC would say about this attempt.

Reply:
So, it's fixed in eyes of CC.